File: state/state_test.go
package state

import "testing"

func TestWeek(t *testing.T) {
	ctx := NewDayContext()

	results := []string{"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"}
	for i := 0; i < 8; i++ {
		if ctx.Today() != results[i%7] {
			t.Fail()
		}
		ctx.Next()
	}
}

File: state/cpp/state_test.cc
#include <gtest/gtest.h>
#include "state/cpp/state.h"

TEST(TestState, TestDayState) {
    DayContext ctx = DayContext();
     const std::string days[7] = {"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"};
     for(int i=0; i<8; ++i) {
        EXPECT_EQ(ctx.Today(), days[i%7]);
        ctx.Next();
     }
}
